docs/modules/ROOT/pages/user-guide/cdi.adoc

```diff
@@ -178,6 +178,41 @@ public class CamelRoute extends RouteBuilder {
 }
 ----
 
+As an alternative to `@Named`, you may also use `io.smallrye.common.annotation.Identifier` to name and identify a bean. Note that the bean is annotated with `@Unremovable`, since `@Identifier` is not a bean defining annotation and thus Quarkus may potentially remove the bean if it considers it unused.
+
```

Comment:
I'm just wondering if we can use `UnremovableBeanBuildItem.beanClassAnnotation(DotName.create(Identifier.class)` to avoid using `@Unremovable`. see https://github.com/quarkusio/quarkus/blob/main/extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/UnremovableBeanBuildItem.java#L140-L150

Reply:
Yeah - I actually had that code originally and for some reason thought it was maybe not a good idea. So I removed it.

But thinking about it more, it's seems fine. So I can put it back again to improve the UX.

Reply:
Any idea if it will be integrated in Quarkus also ?

Reply:
Seems at the movement it's done selectively based on whether extensions expect `@Identifier`. E.g like the SmallRye reactive streams extensions.

Reply:
Ok, so it is really something at Camel Quarkus level and shouldn't clash with Quarkus future :+1: